orcid-message-listener/src/main/java/org/orcid/listener/ReIndexListener.java

```diff
@@ -64,23 +69,49 @@
      * @throws AmazonClientException 
      */
     @JmsListener(destination = MessageConstants.Queues.REINDEX)
-    public void processMessage(final Map<String, String> map) throws JsonProcessingException, AmazonClientException, JAXBException {
+    public void processMessage(final Map<String, String> map) throws JsonProcessingException, AmazonClientException, JAXBException {        
         LastModifiedMessage message = new LastModifiedMessage(map);
-        LOG.info("Recieved " + MessageConstants.Queues.REINDEX + " message for orcid " + message.getOrcid() + " " + message.getLastUpdated());
-
-        try{
-            OrcidMessage profile = orcid12ApiClient.fetchPublicProfile(message.getOrcid());
-            Record record = orcid20ApiClient.fetchPublicProfile(message.getOrcid());//can we not just transform the above?
+        String orcid = message.getOrcid();
+        LOG.info("Recieved " + MessageConstants.Queues.REINDEX + " message for orcid " + orcid + " " + message.getLastUpdated());
+        OrcidMessage profile = null;
+        Record record = null;
+        
+        //If record is locked
+        try {
+            profile = orcid12ApiClient.fetchPublicProfile(orcid);
+        } catch(LockedRecordException lre) {
+            try {
+                exceptionHandler.handleLockedRecordException(message, lre.getOrcidMessage());                
+            } catch (DeprecatedRecordException e) {
+                // Should never happen, since it is already locked
+            }
+            return; 
+        } catch (DeprecatedRecordException dre) {
+            try {
+                exceptionHandler.handleDeprecatedRecordException(message, dre.getOrcidDeprecated());
+            } catch (LockedRecordException e) {
+                // Should never happen, since it is already deprecated
+            }
+            return;
+        }
+                
+        //Fetch 2.0 record
+        try {
+            record = orcid20ApiClient.fetchPublicProfile(orcid);
+        } catch(Exception e) {
+            LOG.warn("Unable to fetch record " + orcid + " from 2.0 API");            
```

Comment:
Warn for now, on this phase (phase 1), if we where able to process the 1.2 record, we are fine, so, if something wrong happens fetching 2.0 we just log it as a warn.

On phase 2 the idea is to have some place (a DB) to write an error and a status to indicate the record was not sent to S3, but, for phase 1 is ok to just log a warn message